src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -380,6 +380,56 @@ public void TestDefaultParentContext()
             }).Dispose();
         }
 
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestCreatingActivityUsingDifferentParentIds()
+        {
+            RemoteExecutor.Invoke(() => {
+
+                int callingByContext = 0;
+                int callingByParentId = 0;
+
+                using ActivitySource aSource = new ActivitySource("ParentIdsTest");
+                using ActivityListener listener1 = new ActivityListener();  // will have context callback only
+                using ActivityListener listener2 = new ActivityListener();  // will have parent id  callback only
+                using ActivityListener listener3 = new ActivityListener();  // will have both context and parent Id callbacks
+
+                listener1.ShouldListenTo = listener2.ShouldListenTo = listener3.ShouldListenTo = (activitySource) => activitySource.Name == "ParentIdsTest";
+                listener1.GetRequestedDataUsingContext = listener3.GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) =>
+                {
+                    callingByContext++;
```

Comment:
Do you also need to validate the the activityOptions.Parent.Context is matching the passed parentid - const string w3cId = "00-99d43cb30a4cdb4fbeee3a19c29201b0-e82825765f051b47-00";

Reply:
good idea. I'll do that.